Model/Menu.php

```diff
@@ -36,6 +64,10 @@ public function getStores()
 
     public function saveStores(array $stores)
     {
+        if ($stores == $this->getStores()) {
```

Comment:
Is this really necessary? I have seen a few lines below that all entries related to the current menu are removed from the table and then they're reinserted: https://github.com/SnowdogApps/magento2-menu/blob/develop/Model/Menu.php#L42

Reply:
I added this in order to avoid running unnecessary `delete` SQL query and the `for` loop `insert` queries (which can be refactored into `insertMultiple`) when there are no store changes.  
I also added it to avoid invalidating page cache [here](https://github.com/SnowdogApps/magento2-menu/blob/837ce8d2ed355bc8e19f0b61e595742b10b503db/Model/Menu.php#L79) unnecessarily, when there are no stores data changes.

The only downside to it is that if there are actually store changes, then it will run an additional `select` SQL query beside the delete and insert queries.  
However, I think it's worth it, as we should not delete and reinsert unchanged stores records,  
whenever we are just editing menu nodes for example (Which is probably the most thing to do when editing a menu).

Reply:
Ok, it sounds logic and I think we can keep it, thanks